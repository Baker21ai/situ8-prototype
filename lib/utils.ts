// Re-export utility functions for consistency across the application
export { cn } from '../components/ui/utils';