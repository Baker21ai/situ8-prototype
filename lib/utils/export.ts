/**
 * Export utilities for activities data
 * Supports CSV and JSON export formats
 */

import { EnterpriseActivity } from '../types/activity';

export interface ExportOptions {
  format: 'csv' | 'json';
  filename?: string;
  includeFields?: string[];
  dateRange?: {
    start: Date;
    end: Date;
  };
  filterBy?: {
    priority?: string;
    status?: string;
    assignedTo?: string;
  };
}

// Default fields for export
const DEFAULT_EXPORT_FIELDS = [
  'id',
  'title',
  'type',
  'priority',
  'status',
  'timestamp',
  'location',
  'description',
  'assignedTo',
  'confidence'
];

/**
 * Export activities to CSV format
 */
export function exportToCSV(activities: EnterpriseActivity[], options: ExportOptions = { format: 'csv' }): string {
  const fields = options.includeFields || DEFAULT_EXPORT_FIELDS;
  const filteredActivities = filterActivities(activities, options);
  
  // Create CSV header
  const header = fields.join(',');
  
  // Create CSV rows
  const rows = filteredActivities.map(activity => {
    return fields.map(field => {
      const value = getFieldValue(activity, field);
      // Escape commas and quotes for CSV
      if (typeof value === 'string' && (value.includes(',') || value.includes('"'))) {
        return `"${value.replace(/"/g, '""')}"`;
      }
      return value;
    }).join(',');
  });
  
  return [header, ...rows].join('\n');
}

/**
 * Export activities to JSON format
 */
export function exportToJSON(activities: EnterpriseActivity[], options: ExportOptions = { format: 'json' }): string {
  const fields = options.includeFields || DEFAULT_EXPORT_FIELDS;
  const filteredActivities = filterActivities(activities, options);
  
  const exportData = filteredActivities.map(activity => {
    const exportActivity: any = {};
    fields.forEach(field => {
      exportActivity[field] = getFieldValue(activity, field);
    });
    return exportActivity;
  });
  
  return JSON.stringify({
    exportInfo: {
      timestamp: new Date().toISOString(),
      totalRecords: exportData.length,
      filters: options.filterBy || {},
      dateRange: options.dateRange
    },
    activities: exportData
  }, null, 2);
}

/**
 * Download exported data as file
 */
export function downloadExport(data: string, options: ExportOptions): void {
  const { format, filename } = options;
  const timestamp = new Date().toISOString().split('T')[0];
  const defaultFilename = `situ8-activities-${timestamp}.${format}`;
  
  const blob = new Blob([data], {
    type: format === 'csv' ? 'text/csv;charset=utf-8;' : 'application/json;charset=utf-8;'
  });
  
  const link = document.createElement('a');
  const url = URL.createObjectURL(blob);
  link.setAttribute('href', url);
  link.setAttribute('download', filename || defaultFilename);
  link.style.visibility = 'hidden';
  document.body.appendChild(link);
  link.click();
  document.body.removeChild(link);
  URL.revokeObjectURL(url);
}

/**
 * Main export function that handles both formats
 */
export function exportActivities(activities: EnterpriseActivity[], options: ExportOptions): void {
  let exportData: string;
  
  if (options.format === 'csv') {
    exportData = exportToCSV(activities, options);
  } else {
    exportData = exportToJSON(activities, options);
  }
  
  downloadExport(exportData, options);
}

/**
 * Get export statistics
 */
export function getExportStats(activities: EnterpriseActivity[], options: ExportOptions) {
  const filteredActivities = filterActivities(activities, options);
  
  const stats = {
    totalActivities: filteredActivities.length,
    priorityBreakdown: {} as Record<string, number>,
    statusBreakdown: {} as Record<string, number>,
    typeBreakdown: {} as Record<string, number>,
    dateRange: {
      earliest: null as Date | null,
      latest: null as Date | null
    }
  };
  
  filteredActivities.forEach(activity => {
    // Priority breakdown
    const priority = activity.priority || 'unknown';
    stats.priorityBreakdown[priority] = (stats.priorityBreakdown[priority] || 0) + 1;
    
    // Status breakdown
    const status = activity.status || 'unknown';
    stats.statusBreakdown[status] = (stats.statusBreakdown[status] || 0) + 1;
    
    // Type breakdown
    const type = activity.type || 'unknown';
    stats.typeBreakdown[type] = (stats.typeBreakdown[type] || 0) + 1;
    
    // Date range
    if (activity.timestamp) {
      const date = new Date(activity.timestamp);
      if (!stats.dateRange.earliest || date < stats.dateRange.earliest) {
        stats.dateRange.earliest = date;
      }
      if (!stats.dateRange.latest || date > stats.dateRange.latest) {
        stats.dateRange.latest = date;
      }
    }
  });
  
  return stats;
}

// Helper functions

function filterActivities(activities: EnterpriseActivity[], options: ExportOptions): EnterpriseActivity[] {
  let filtered = [...activities];
  
  // Filter by date range
  if (options.dateRange) {
    filtered = filtered.filter(activity => {
      if (!activity.timestamp) return false;
      const date = new Date(activity.timestamp);
      return date >= options.dateRange!.start && date <= options.dateRange!.end;
    });
  }
  
  // Filter by criteria
  if (options.filterBy) {
    const { priority, status, assignedTo } = options.filterBy;
    
    if (priority) {
      filtered = filtered.filter(activity => activity.priority === priority);
    }
    
    if (status) {
      filtered = filtered.filter(activity => activity.status === status);
    }
    
    if (assignedTo) {
      filtered = filtered.filter(activity => activity.assignedTo === assignedTo);
    }
  }
  
  return filtered;
}

function getFieldValue(activity: EnterpriseActivity, field: string): string {
  const value = (activity as any)[field];
  
  if (value === null || value === undefined) {
    return '';
  }
  
  if (value instanceof Date) {
    return value.toISOString();
  }
  
  if (typeof value === 'object') {
    return JSON.stringify(value);
  }
  
  return String(value);
}

// Predefined export templates
export const EXPORT_TEMPLATES = {
  basic: {
    name: 'Basic Report',
    fields: ['id', 'title', 'priority', 'status', 'timestamp', 'location', 'assignedTo']
  },
  detailed: {
    name: 'Detailed Report',
    fields: DEFAULT_EXPORT_FIELDS
  },
  investigation: {
    name: 'Investigation Report',
    fields: ['id', 'title', 'type', 'priority', 'timestamp', 'location', 'description', 'confidence', 'assignedTo']
  },
  summary: {
    name: 'Summary Report',
    fields: ['id', 'title', 'priority', 'status', 'timestamp', 'assignedTo']
  }
};