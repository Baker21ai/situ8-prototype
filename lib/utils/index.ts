/**
 * Central export for all utility functions
 */

// Time utilities
export * from './time';

// Status utilities
export * from './status';

// Security utilities
export * from './security';

// Re-export cn utility from UI components
export { cn } from '@/components/ui/utils';

// Common utility functions
export const noop = () => {};

export const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));

export const debounce = <T extends (...args: any[]) => any>(
  func: T,
  wait: number
): ((...args: Parameters<T>) => void) => {
  let timeout: NodeJS.Timeout;
  return (...args: Parameters<T>) => {
    clearTimeout(timeout);
    timeout = setTimeout(() => func(...args), wait);
  };
};

export const throttle = <T extends (...args: any[]) => any>(
  func: T,
  limit: number
): ((...args: Parameters<T>) => void) => {
  let inThrottle: boolean;
  return (...args: Parameters<T>) => {
    if (!inThrottle) {
      func(...args);
      inThrottle = true;
      setTimeout(() => inThrottle = false, limit);
    }
  };
};

export const capitalize = (str: string): string => {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

export const truncate = (str: string, length: number): string => {
  if (str.length <= length) return str;
  return str.slice(0, length) + '...';
};

export const formatBytes = (bytes: number, decimals = 2): string => {
  if (bytes === 0) return '0 Bytes';
  
  const k = 1024;
  const dm = decimals < 0 ? 0 : decimals;
  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
  
  const i = Math.floor(Math.log(bytes) / Math.log(k));
  
  return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
};

export const generateId = (prefix = 'id'): string => {
  return `${prefix}_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`;
};

export const groupBy = <T>(array: T[], key: keyof T): Record<string, T[]> => {
  return array.reduce((result, item) => {
    const group = String(item[key]);
    if (!result[group]) result[group] = [];
    result[group].push(item);
    return result;
  }, {} as Record<string, T[]>);
};

export const sortBy = <T>(array: T[], key: keyof T, order: 'asc' | 'desc' = 'asc'): T[] => {
  return [...array].sort((a, b) => {
    const aVal = a[key];
    const bVal = b[key];
    
    if (aVal < bVal) return order === 'asc' ? -1 : 1;
    if (aVal > bVal) return order === 'asc' ? 1 : -1;
    return 0;
  });
};

export const chunk = <T>(array: T[], size: number): T[][] => {
  const chunks: T[][] = [];
  for (let i = 0; i < array.length; i += size) {
    chunks.push(array.slice(i, i + size));
  }
  return chunks;
};

export const unique = <T>(array: T[], key?: keyof T): T[] => {
  if (!key) {
    return [...new Set(array)];
  }
  
  const seen = new Set();
  return array.filter(item => {
    const val = item[key];
    if (seen.has(val)) return false;
    seen.add(val);
    return true;
  });
};

export const deepClone = <T>(obj: T): T => {
  return JSON.parse(JSON.stringify(obj));
};

export const isEmptyObject = (obj: object): boolean => {
  return Object.keys(obj).length === 0;
};

export const pick = <T, K extends keyof T>(obj: T, keys: K[]): Pick<T, K> => {
  const result = {} as Pick<T, K>;
  keys.forEach(key => {
    if (key in obj) {
      result[key] = obj[key];
    }
  });
  return result;
};

export const omit = <T, K extends keyof T>(obj: T, keys: K[]): Omit<T, K> => {
  const result = { ...obj };
  keys.forEach(key => {
    delete result[key];
  });
  return result as Omit<T, K>;
};