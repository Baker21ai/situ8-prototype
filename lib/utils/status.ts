/**
 * Status and priority utility functions for consistent styling across the application
 */

export type Priority = 'critical' | 'high' | 'medium' | 'low';
export type Status = 'detecting' | 'assigned' | 'responding' | 'resolved';
export type BusinessImpact = 'none' | 'low' | 'medium' | 'high' | 'critical';

export interface ColorSet {
  border: string;
  background: string;
  text: string;
  icon?: string;
  ring?: string;
}

/**
 * Get consistent color classes for priority levels
 * @param priority - Priority level
 * @param variant - Style variant ('default' | 'compact' | 'minimal')
 * @returns Object with Tailwind classes for styling
 */
export function getPriorityColor(priority: Priority, variant: 'default' | 'compact' | 'minimal' = 'default'): ColorSet {
  const colors: Record<Priority, ColorSet> = {
    critical: {
      border: 'border-red-500',
      background: 'bg-red-50',
      text: 'text-red-800',
      icon: 'text-red-600',
      ring: 'ring-red-500'
    },
    high: {
      border: 'border-orange-500',
      background: 'bg-orange-50',
      text: 'text-orange-800',
      icon: 'text-orange-600',
      ring: 'ring-orange-500'
    },
    medium: {
      border: 'border-yellow-500',
      background: 'bg-yellow-50',
      text: 'text-yellow-800',
      icon: 'text-yellow-600',
      ring: 'ring-yellow-500'
    },
    low: {
      border: 'border-green-500',
      background: 'bg-green-50',
      text: 'text-green-800',
      icon: 'text-green-600',
      ring: 'ring-green-500'
    }
  };

  if (variant === 'minimal') {
    // Return only essential colors for minimal displays
    return {
      border: colors[priority].border,
      background: colors[priority].background,
      text: colors[priority].text
    };
  }

  return colors[priority] || colors.low;
}

/**
 * Get consistent color classes for status values
 * @param status - Status value
 * @returns Object with Tailwind classes for styling
 */
export function getStatusColor(status: Status): ColorSet {
  const colors: Record<Status, ColorSet> = {
    detecting: {
      border: 'border-blue-500',
      background: 'bg-blue-50',
      text: 'text-blue-800',
      icon: 'text-blue-600'
    },
    assigned: {
      border: 'border-orange-500',
      background: 'bg-orange-50',
      text: 'text-orange-800',
      icon: 'text-orange-600'
    },
    responding: {
      border: 'border-yellow-500',
      background: 'bg-yellow-50',
      text: 'text-yellow-800',
      icon: 'text-yellow-600'
    },
    resolved: {
      border: 'border-green-500',
      background: 'bg-green-50',
      text: 'text-green-800',
      icon: 'text-green-600'
    }
  };

  return colors[status] || colors.resolved;
}

/**
 * Get color for business impact levels
 * @param impact - Business impact level
 * @returns Tailwind color class
 */
export function getBusinessImpactColor(impact: BusinessImpact): string {
  const colors: Record<BusinessImpact, string> = {
    critical: 'bg-red-500',
    high: 'bg-orange-500',
    medium: 'bg-yellow-500',
    low: 'bg-blue-500',
    none: 'bg-gray-400'
  };

  return colors[impact] || colors.none;
}

/**
 * Get priority display properties
 * @param priority - Priority level
 * @returns Object with display properties
 */
export function getPriorityDisplay(priority: Priority) {
  const displays: Record<Priority, { label: string; icon: string; pulse: boolean }> = {
    critical: {
      label: 'CRITICAL',
      icon: 'ðŸ”´',
      pulse: true
    },
    high: {
      label: 'HIGH',
      icon: 'ðŸŸ ',
      pulse: false
    },
    medium: {
      label: 'MEDIUM',
      icon: 'ðŸŸ¡',
      pulse: false
    },
    low: {
      label: 'LOW',
      icon: 'ðŸŸ¢',
      pulse: false
    }
  };

  return displays[priority] || displays.low;
}

/**
 * Get status display properties
 * @param status - Status value
 * @returns Object with display properties
 */
export function getStatusDisplay(status: Status) {
  const displays: Record<Status, { label: string; description: string }> = {
    detecting: {
      label: 'DETECTING',
      description: 'Initial detection phase'
    },
    assigned: {
      label: 'ASSIGNED',
      description: 'Assigned to personnel'
    },
    responding: {
      label: 'RESPONDING',
      description: 'Personnel responding to incident'
    },
    resolved: {
      label: 'RESOLVED',
      description: 'Issue resolved'
    }
  };

  return displays[status] || displays.resolved;
}

/**
 * Convert priority to numeric value for sorting
 * @param priority - Priority level
 * @returns Numeric value (higher = more urgent)
 */
export function getPriorityWeight(priority: Priority): number {
  const weights: Record<Priority, number> = {
    critical: 4,
    high: 3,
    medium: 2,
    low: 1
  };

  return weights[priority] || 0;
}

/**
 * Sort items by priority (highest first)
 * @param items - Array of items with priority property
 * @returns Sorted array
 */
export function sortByPriority<T extends { priority: Priority }>(items: T[]): T[] {
  return [...items].sort((a, b) => {
    return getPriorityWeight(b.priority) - getPriorityWeight(a.priority);
  });
}

/**
 * Get combined status and priority classes
 * @param priority - Priority level
 * @param status - Status value
 * @returns Combined Tailwind classes
 */
export function getActivityClasses(priority: Priority, status: Status) {
  const priorityColors = getPriorityColor(priority);
  const isActive = status === 'detecting' || status === 'responding';
  
  return {
    container: `${priorityColors.border} ${priorityColors.background} ${isActive ? 'animate-pulse' : ''}`,
    badge: `${priorityColors.background} ${priorityColors.text} ${priorityColors.border}`,
    icon: priorityColors.icon || '',
    text: priorityColors.text
  };
}