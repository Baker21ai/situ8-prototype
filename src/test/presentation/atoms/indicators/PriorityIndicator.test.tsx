import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { 
  PriorityIndicator, 
  PriorityIndicatorIcon,
  PriorityIndicatorDot,
  PriorityIndicatorText,
  PriorityIndicatorCombined,
  PriorityIndicatorProps 
} from '../../../../presentation/atoms/indicators/PriorityIndicator';
import { Priority } from '../../../../../lib/utils/status';

// Mock the Badge component
vi.mock('../../../../../components/ui/badge', () => ({
  Badge: ({ children, className, ...props }: any) => (
    <div data-testid="priority-badge" className={className} {...props}>
      {children}
    </div>
  )
}));

// Mock Lucide icons
vi.mock('lucide-react', () => ({
  AlertTriangle: ({ className, ...props }: any) => (
    <div data-testid="alert-triangle-icon" className={className} {...props} />
  ),
  AlertCircle: ({ className, ...props }: any) => (
    <div data-testid="alert-circle-icon" className={className} {...props} />
  ),
  Info: ({ className, ...props }: any) => (
    <div data-testid="info-icon" className={className} {...props} />
  ),
  CheckCircle: ({ className, ...props }: any) => (
    <div data-testid="check-circle-icon" className={className} {...props} />
  ),
}));

describe('PriorityIndicator', () => {
  const defaultProps: PriorityIndicatorProps = {
    priority: 'medium'
  };

  describe('Rendering', () => {
    it('should render with default props', () => {
      render(<PriorityIndicator {...defaultProps} />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator).toBeInTheDocument();
      expect(indicator).toHaveTextContent('Medium');
    });

    it('should render all priority levels correctly', () => {
      const priorities: Priority[] = ['critical', 'high', 'medium', 'low'];
      const expectedLabels = ['Critical', 'High', 'Medium', 'Low'];
      
      priorities.forEach((priority, index) => {
        const { rerender } = render(<PriorityIndicator priority={priority} />);
        
        const indicator = screen.getByTestId('priority-badge');
        expect(indicator).toHaveTextContent(expectedLabels[index]);
        
        rerender(<div />);
      });
    });
  });

  describe('Priority-based styling', () => {
    it('should apply critical priority styles', () => {
      render(<PriorityIndicator priority="critical" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('bg-red-500/20');
      expect(indicator.className).toContain('border-red-500');
      expect(indicator.className).toContain('text-red-300');
    });

    it('should apply high priority styles', () => {
      render(<PriorityIndicator priority="high" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('bg-orange-500/20');
      expect(indicator.className).toContain('border-orange-500');
      expect(indicator.className).toContain('text-orange-300');
    });

    it('should apply medium priority styles', () => {
      render(<PriorityIndicator priority="medium" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('bg-yellow-500/20');
      expect(indicator.className).toContain('border-yellow-500');
      expect(indicator.className).toContain('text-yellow-300');
    });

    it('should apply low priority styles', () => {
      render(<PriorityIndicator priority="low" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('bg-green-500/20');
      expect(indicator.className).toContain('border-green-500');
      expect(indicator.className).toContain('text-green-300');
    });
  });

  describe('Size variants', () => {
    it('should apply extra small size styles', () => {
      render(<PriorityIndicator {...defaultProps} size="xs" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('text-xs');
      expect(indicator.className).toContain('px-1');
      expect(indicator.className).toContain('py-0.5');
    });

    it('should apply small size styles (default)', () => {
      render(<PriorityIndicator {...defaultProps} size="sm" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('text-xs');
      expect(indicator.className).toContain('px-1.5');
      expect(indicator.className).toContain('py-0.5');
    });

    it('should apply medium size styles', () => {
      render(<PriorityIndicator {...defaultProps} size="md" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('text-sm');
      expect(indicator.className).toContain('px-2');
      expect(indicator.className).toContain('py-1');
    });

    it('should apply large size styles', () => {
      render(<PriorityIndicator {...defaultProps} size="lg" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('text-base');
      expect(indicator.className).toContain('px-3');
      expect(indicator.className).toContain('py-1.5');
    });

    it('should apply extra large size styles', () => {
      render(<PriorityIndicator {...defaultProps} size="xl" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('text-lg');
      expect(indicator.className).toContain('px-4');
      expect(indicator.className).toContain('py-2');
    });
  });

  describe('Variant types', () => {
    it('should render badge variant (default)', () => {
      render(<PriorityIndicator {...defaultProps} variant="badge" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator).toBeInTheDocument();
      expect(indicator).toHaveTextContent('Medium');
    });

    it('should render icon variant with icon and label', () => {
      render(<PriorityIndicator priority="critical" variant="icon" />);
      
      const icon = screen.getByTestId('alert-triangle-icon');
      const text = screen.getByText('Critical');
      
      expect(icon).toBeInTheDocument();
      expect(text).toBeInTheDocument();
    });

    it('should render icon variant without label when showLabel is false', () => {
      render(<PriorityIndicator priority="critical" variant="icon" showLabel={false} />);
      
      const icon = screen.getByTestId('alert-triangle-icon');
      expect(icon).toBeInTheDocument();
      expect(screen.queryByText('Critical')).not.toBeInTheDocument();
    });

    it('should render dot variant', () => {
      render(<PriorityIndicator priority="high" variant="dot" />);
      
      const container = screen.getByText('High').parentElement;
      expect(container).toBeInTheDocument();
      
      // Check for dot element (div with rounded-full)
      const dotElement = container?.querySelector('div[class*="rounded-full"]');
      expect(dotElement).toBeInTheDocument();
    });

    it('should render text variant', () => {
      render(<PriorityIndicator priority="low" variant="text" />);
      
      const textElement = screen.getByText('Low');
      expect(textElement).toBeInTheDocument();
      expect(textElement.className).toContain('font-medium');
      expect(textElement.className).toContain('uppercase');
    });

    it('should render combined variant with icon and text', () => {
      render(<PriorityIndicator priority="medium" variant="combined" />);
      
      const indicator = screen.getByTestId('priority-badge');
      const icon = screen.getByTestId('info-icon');
      
      expect(indicator).toBeInTheDocument();
      expect(icon).toBeInTheDocument();
      expect(indicator).toHaveTextContent('Medium');
    });
  });

  describe('Animation', () => {
    it('should apply pulse animation when animated is true', () => {
      render(<PriorityIndicator {...defaultProps} animated={true} />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain('animate-pulse');
    });

    it('should apply pulse animation for critical priority by default', () => {
      render(<PriorityIndicator priority="critical" />);
      
      const container = screen.getByText('Critical').parentElement;
      expect(container?.className).toContain('animate-pulse');
    });

    it('should not apply pulse animation for non-critical priorities by default', () => {
      render(<PriorityIndicator priority="low" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).not.toContain('animate-pulse');
    });
  });

  describe('Custom styling', () => {
    it('should apply custom className', () => {
      const customClass = 'custom-priority-class';
      render(<PriorityIndicator {...defaultProps} className={customClass} />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain(customClass);
    });

    it('should combine custom className with default styles', () => {
      const customClass = 'border-4';
      render(<PriorityIndicator {...defaultProps} className={customClass} />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator.className).toContain(customClass);
      expect(indicator.className).toContain('border');
      expect(indicator.className).toContain('font-medium');
    });
  });

  describe('Icon mapping', () => {
    it('should use AlertTriangle icon for critical priority', () => {
      render(<PriorityIndicator priority="critical" variant="icon" />);
      expect(screen.getByTestId('alert-triangle-icon')).toBeInTheDocument();
    });

    it('should use AlertCircle icon for high priority', () => {
      render(<PriorityIndicator priority="high" variant="icon" />);
      expect(screen.getByTestId('alert-circle-icon')).toBeInTheDocument();
    });

    it('should use Info icon for medium priority', () => {
      render(<PriorityIndicator priority="medium" variant="icon" />);
      expect(screen.getByTestId('info-icon')).toBeInTheDocument();
    });

    it('should use CheckCircle icon for low priority', () => {
      render(<PriorityIndicator priority="low" variant="icon" />);
      expect(screen.getByTestId('check-circle-icon')).toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('should have proper aria-label for icon variant', () => {
      render(<PriorityIndicator priority="critical" variant="icon" />);
      
      const icon = screen.getByTestId('alert-triangle-icon');
      expect(icon).toHaveAttribute('aria-label', 'Critical priority');
    });

    it('should have proper aria-label for dot variant', () => {
      render(<PriorityIndicator priority="high" variant="dot" />);
      
      const container = screen.getByText('High').parentElement;
      const dotElement = container?.querySelector('div[aria-label="High priority"]');
      expect(dotElement).toBeInTheDocument();
    });

    it('should have proper aria-label for text variant', () => {
      render(<PriorityIndicator priority="low" variant="text" />);
      
      const textElement = screen.getByText('Low');
      expect(textElement).toHaveAttribute('aria-label', 'Low priority');
    });

    it('should be accessible with screen readers', () => {
      render(<PriorityIndicator priority="critical" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator).toHaveTextContent('Critical');
      expect(indicator).toBeVisible();
    });
  });

  describe('Specialized variants', () => {
    it('should render PriorityIndicatorIcon correctly', () => {
      render(<PriorityIndicatorIcon priority="critical" />);
      
      const icon = screen.getByTestId('alert-triangle-icon');
      const text = screen.getByText('Critical');
      
      expect(icon).toBeInTheDocument();
      expect(text).toBeInTheDocument();
    });

    it('should render PriorityIndicatorDot correctly', () => {
      render(<PriorityIndicatorDot priority="high" />);
      
      const container = screen.getByText('High').parentElement;
      const dotElement = container?.querySelector('div[class*="rounded-full"]');
      expect(dotElement).toBeInTheDocument();
    });

    it('should render PriorityIndicatorText correctly', () => {
      render(<PriorityIndicatorText priority="medium" />);
      
      const textElement = screen.getByText('Medium');
      expect(textElement).toBeInTheDocument();
      expect(textElement.className).toContain('uppercase');
    });

    it('should render PriorityIndicatorCombined correctly', () => {
      render(<PriorityIndicatorCombined priority="low" />);
      
      const indicator = screen.getByTestId('priority-badge');
      const icon = screen.getByTestId('check-circle-icon');
      
      expect(indicator).toBeInTheDocument();
      expect(icon).toBeInTheDocument();
      expect(indicator).toHaveTextContent('Low');
    });
  });

  describe('Edge cases', () => {
    it('should handle empty className', () => {
      render(<PriorityIndicator {...defaultProps} className="" />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator).toBeInTheDocument();
    });

    it('should handle undefined className', () => {
      render(<PriorityIndicator {...defaultProps} className={undefined} />);
      
      const indicator = screen.getByTestId('priority-badge');
      expect(indicator).toBeInTheDocument();
    });

    it('should work with all combinations of props', () => {
      const priorities: Priority[] = ['critical', 'high', 'medium', 'low'];
      const variants = ['badge', 'icon', 'dot', 'text', 'combined'] as const;
      const sizes = ['xs', 'sm', 'md', 'lg', 'xl'] as const;
      
      priorities.forEach(priority => {
        variants.forEach(variant => {
          sizes.forEach(size => {
            const { rerender } = render(
              <PriorityIndicator 
                priority={priority} 
                variant={variant}
                size={size}
                animated={priority === 'critical'}
                showLabel={true}
              />
            );
            
            // Basic smoke test - component should render without errors
            if (variant === 'badge' || variant === 'combined') {
              expect(screen.getByTestId('priority-badge')).toBeInTheDocument();
            }
            
            rerender(<div />);
          });
        });
      });
    });
  });

  describe('Snapshot tests', () => {
    it('should match snapshot for default variant', () => {
      const { container } = render(<PriorityIndicator priority="medium" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for icon variant', () => {
      const { container } = render(<PriorityIndicator priority="critical" variant="icon" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for dot variant', () => {
      const { container } = render(<PriorityIndicator priority="high" variant="dot" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
