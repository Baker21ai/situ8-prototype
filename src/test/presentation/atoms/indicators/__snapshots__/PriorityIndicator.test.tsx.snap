// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PriorityIndicator > Snapshot tests > should match snapshot for default variant 1`] = `
<div
  class="
            px-1.5 py-0.5 text-xs
            bg-yellow-500/20 text-yellow-300 border-yellow-500
            border font-medium 
    
    
  
          "
  data-testid="priority-badge"
>
  Medium
</div>
`;

exports[`PriorityIndicator > Snapshot tests > should match snapshot for dot variant 1`] = `
<div
  class="inline-flex items-center gap-1 
    
    
  "
>
  <div
    aria-label="High priority"
    class="rounded-full w-2.5 h-2.5 bg-orange-500"
  />
  <span
    class="text-xs text-orange-300 font-medium"
  >
    High
  </span>
</div>
`;

exports[`PriorityIndicator > Snapshot tests > should match snapshot for icon variant 1`] = `
<div
  class="inline-flex items-center gap-1 
    animate-pulse
    
  "
>
  <div
    aria-label="Critical priority"
    class="h-3 w-3 text-red-400"
    data-testid="alert-triangle-icon"
  />
  <span
    class="text-xs text-red-300 font-medium"
  >
    Critical
  </span>
</div>
`;
